controller/array_action/array_mediator_ds8k.py

```diff
@@ -427,24 +424,21 @@ def _get_api_volume_by_name(self, volume_name, pool_id):
                 "pool_id is not specified, can not get volumes from storage."
             )
             raise array_errors.PoolParameterIsMissing(self.array_type)
-        else:
-            pools = [Munch({"id": pool_id})]
-
-        for pool in pools:
-            volume_candidates = []
-            try:
-                volume_candidates.extend(self.client.get_volumes_by_pool(pool.id))
-            except (exceptions.NotFound, exceptions.InternalServerError) as ex:
-                if ERROR_CODE_RESOURCE_NOT_EXISTS or INCORRECT_ID in str(ex.message).upper():
-                    raise array_errors.PoolDoesNotExist(pool.id, self.identifier)
-                else:
-                    raise ex
-            for volume in volume_candidates:
-                logger.info("Checking volume: {}".format(volume.name))
-                if volume.name == volume_name:
-                    logger.debug("Found volume: {}".format(volume))
-                    volume.flashcopy = self.get_flashcopies_by_volume(volume.id)
-                    return volume
+
+        volume_candidates = []
```

Comment:
if `except` always `raise`s, why not just define `volume_candidates` inside `try`?

Reply:
done